src/creature.h

```diff
@@ -408,6 +408,14 @@ class Creature : virtual public Thing
 		void setUseDefense(bool useDefense) {
 			canUseDefense = useDefense;
 		}
+		void setCanMove(bool move)
+		{
+			canMove = move;
+			cancelNextWalk = true;
```

Comment:
if we set canMove to true, do we still cancelNextWalk?

Reply:
I too was wondering about cancelNextWalk  always being set to true in this function. What is the reason for the code not being this or similar
```suggestion
		void setCanMove(bool move)
		{
			canMove = move;
			cancelNextWalk = !move;
```

Reply:
I'm still curious about this.

Reply:
My guess is to fix desync issues mid-action, this is called to toggle the states, but no matter which direction it toggles to, you dont want to give the user half a sqm headstart as it may create odd client behavior. So even if you just got permission to walk, the character pos gets a cancel byte one last time to set the timings straight.

Reply:
@Znote but if player is already movement blocked then he cannot walk, so what desynchro?